feat(member) : 회원 도메인등록

loustler opened the pull request:
- 회원 도메인 등록
- Level은 임의 포인트 산정으로 처리(enum 사용구분)

Reference #5

각 Level별 포인트는 기준이 필요할 것 같습니다.

loustler committed:
feat(member) : 회원 도메인등록

- 회원 도메인 등록
- Level은 임의 포인트 산정으로 처리(enum 사용구분)

Reference #5

devsejong commented on src/main/java/io/codechobo/domain/member/Level.java:
```diff
@@ -0,0 +1,76 @@
+    private Integer seq;
+
+    @Column(name = "LEVEL_POINT")
+    private Integer point; // 유저가 적립한 포인트
```
등급에 포인트가 들어가는건 좋지않은 것 같네요.
유저가 이 정보를 가지는게 더 옳지 않을까요?

loustler replied:
@devsejong 
member가 포인트를 가지는 게 나을까요?
member하고 완전히 분리하려고 level을 분리했는데, 
다시 합치면 음.. 
level을 그냥 member가 가지고 있는 게 나으려나요?

devsejong replied:
주문을 할때에 포인트를 사용한다면, "회원의 등급에서 포인트를 사용한다"가 되어버려서요. 이렇게 되면 의미상으로 맞지 않아서요.

"회원의 포인트를 사용한다"가 더 어울릴 듯 하고, 회원이 포인트를 직접 가지고 있는게 좋을 듯 합니다.

loustler replied:
@devsejong 
다양한 관측에서 보면 그렇네요.

회원의 등급에서 포인트를...? 

말씀하신 대로 의미가 이상하네요.

Level은 Member로 귀속시키고 Level쪽은 삭제했어요.

Test 후에 다시 push하겠습니다 :)

devsejong commented on src/main/java/io/codechobo/domain/member/Member.java:
```diff
@@ -0,0 +1,87 @@
+    private String id; // Id Uniq
+
+    @Column(name = "MEMBER_PASSWORD")
+    private String pwd; // 패스워드
```
(리뷰때 개인성향이 드러나는것 같아서 걱정이긴 합니다....)
각 도메인이 가지는 필드는 약어가 아니라 풀 네임으로 작성하는게 좋을 것 같습니다. 필드명을 약어로 쓸 경우에 가독성이 그렇게 좋지 않아서요.

또한 테이블이름의 Member를 통해서 password 가 멤버에 속한다는 것을 유추할 수 있으므로, PASSWORD로 작성하는게 좋을 것 같습니다.

loustler replied:
@devsejong column을 MEMBER_PASSWORD가 아니라 PASSWORD가 낫다는 말씀이시죠?

devsejong replied:
네 저는 그렇게 생각하는데,  맞다 틀리다 정답이 없는 부분이라서요.
다른분 의견 더 들어보고 싶네요.

loustler replied:
@devsejong 
PASSWORD도 MEMBER_PASSWORD -> PASSWORD로 일단 변경했어요.

선반영 후에 다른 분들 이견이 있으시면 그 때 하면 될 것 같아요

devsejong commented on src/main/java/io/codechobo/domain/member/Member.java:
```diff
@@ -0,0 +1,87 @@
+        this.regDate = new Date();
+    }
+
+    public void setId(String id) {this.id = id;}
```
Lombok이 있어서 이걸로 getter, setter를 만들어주면 좋을 것 같네요.

loustler replied:
@devsejong Lombok으로 getter,setter는 처리했어요 :)

devsejong commented on src/main/java/io/codechobo/domain/member/PointPerLevel.java:
```diff
@@ -0,0 +1,14 @@
+ */
+@Getter
+public enum PointPerLevel {
+    BASIC,      // Basic Level
```
등급별 포인트를 여기에 표기하면 좋을 것 같네요. 예를들어서 BASIC의 경우 0~1000점이라고 한다면
BASIC(0, 1000)로 표현한다면 응집력이 올라갈 수 있을 것 같습니다.

loustler replied:
@devsejong 
별도의 필드를 enum에 넣고 constructor로 해야겠네요? 
저도 그렇게 처음에 생각을 했었는데 이걸 적용을 하는 게 조금 헷갈리더라구요 
enum을 많이 사용안해봐서

devsejong replied:
토비의 스프링 가지고 계신다면, 1권 318페이지 참고해보세요. 아마 비슷한 내용 나올꺼예요.

loustler replied:
@devsejong 
constructor는 만들어서 field에 넣는 것 까지는 해놨었는데 
이걸 다시 똑같은 걸로 리턴하는 걸 생각치 못했네요 
도움 감사합니다 👍

loustler committed:
review : LOBOK적용, Level을 Member로 귀속, PointPerLevel 수정

- LOMBOK 적용
- @devsejong님 제안으로 Level을 Member로 귀속
- PointPerLevel Enum Constructor 및 valueOf 추가
- MemberRepository 간단한 query 추가

close #13

devsejong commented:
저도 리뷰 완료했습니다. 머지할께요.
